src/t_string.c

```diff
@@ -599,17 +612,17 @@ void incrDecrCommand(client *c, long long incr) {
     value += incr;
 
     if (o && o->refcount == 1 && o->encoding == OBJ_ENCODING_INT &&
-        (value < 0 || value >= OBJ_SHARED_INTEGERS) &&
         value >= LONG_MIN && value <= LONG_MAX)
     {
         new = o;
         o->ptr = (void*)((long)value);
     } else {
         new = createStringObjectFromLongLongForValue(value);
         if (o) {
-            dbReplaceValueWithDictEntry(c->db,c->argv[1],new,de);
+            dbReplaceValueWithLink(c->db, c->argv[1], &new, link);
         } else {
-            dbAdd(c->db,c->argv[1],new);
+            /* TODO : Pass bucket */
```

Comment:
what is this TODO for?

Reply:
An optimization to dbAdd by link to save lookup:
```
    dictEntLink link; 
    kvobj *o = lookupKeyWriteWithLink(c->db, c->argv[1], &link);
   ...
   dbAddByLink(c->db, c->argv[1], &new, &link);
```

My next commit goanna add here and few other places this optimization.